megatron/optimizer/optimizer.py

```diff
@@ -406,7 +406,7 @@ def get_loss_scale(self):
 
 
     def reload_model_params(self):
-        self._copy_model_params_to_main_params()
```

Comment:
this shouldn't be removed as the change you are doing is incorrect. Here, the model params need to be copied to optimizer params, but you have changed to do the opposite.